pkg/debug/pprof/pprof.go

```diff
@@ -0,0 +1,25 @@
+//go:build debug
+// +build debug
+
+package pprof
+
+import (
+	"net/http"
+	"net/http/pprof"
+
+	"github.com/gorilla/mux"
+
+	"zotregistry.io/zot/pkg/api/config"
+	"zotregistry.io/zot/pkg/log" //nolint:goimports
+)
+
+func SetupPprofRoutes(conf *config.Config, router *mux.Router, authFunc mux.MiddlewareFunc,
+	log log.Logger,
+) {
+	// router has authn mdw already applied
+	pprofRouter := router.PathPrefix("/_zot/debug/pprof").Subrouter()
+	pprofRouter.Methods(http.MethodGet).Handler(http.HandlerFunc(
```

Comment:
Let's check if the `trace` and `goroutines` profiles are useful in our context as well.